### PR TITLE
Remove use_profile_info_card toggle work

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Now that the ProfileInfoCard is stable and accessibility has reviewed it, we are removing all the ProfileInfoTable based code, the feature toggle that was triggering the usage of one or the other, and the tests around that.
- Updated component-migration-samples script to not use the ProfileInfoTable, had to eslint ignore some sections of that file due to it having purposefully added eslint violations to demonstrate script with.
- remove feature toggle name `profile_use_info_card` from platform code

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/62363

## Testing done

- All tests passing, removed feature toggle based tests used specifically for switching between components.

## Screenshots

No UI work, just removing old code

## What areas of the site does it impact?

Profile

## Acceptance criteria

- [x] ProfileInfoTable component removed
- [x] profile_use_info_card feature toggle removed

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed

### CODE REVIEW

**Review by allisonlu — COMMENTED** [review]
–368 LOC feels good 😌 

- [this file is still importing](https://github.com/department-of-veterans-affairs/vets-website/blob/3c41ccbe022f49ff866a1028804ddb40c44dfdb9/script/component-migration/samples/AdditionalInfo.jsx) and using the `ProfileInfoTable` component, but it's in the `/script/` directory rather than `/src/`. Not sure if this is actively being used anywhere on va.gov
- two unit tests are still using the `profileUseInfoCard` toggle: 
  - ContactInformation.unit.spec.jsx
  - PaymentHistory.unit.spec.jsx

**Comment by adamwhitlock1** [comment]
Thanks for catching those things I missed @allisonlu ! I searched the src folder but didn't even realize the script folder had that in it.

Cleaned up and ready to go once tests run.

**Review by allisonlu — APPROVED** [review]
🚀

File: script/component-migration/samples/AdditionalInfo.jsx [hunk 1]
```diff
@@ -71,9 +73,7 @@ export const Sample3 = ({ className = 'sample' }) => (
 
 export const Sample4 = () => (
   <div className="vads-u-margin-y--3 available-connected-apps">
-    <AdditionalInfo
-      triggerText={`What other third-party apps can I connect to my profile?`}
-    >
+    <AdditionalInfo triggerText="What other third-party apps can I connect to my profile?">
```
Comment (jamigibbs) [inline script/component-migration/samples/AdditionalInfo.jsx:76]: Would it be possible to change this to `va-additional-info` while you're making an update to this file? 🙏🏻  I think it should be a pretty straight forward swap out. You don't even need to import anything for it. 

https://design.va.gov/storybook/?path=/docs/components-va-additional-info--default

```js
<va-additional-info trigger="What other third-party apps can I connect to my profile?">
[slot content]
</va-additional-info>
```

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: script/component-migration/samples/AdditionalInfo.jsx [hunk 2]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
 import React from 'react';
+// eslint-disable-next-line deprecate/import
```
[the remark this answers is not in the document]
Comment (jamigibbs) [reply script/component-migration/samples/AdditionalInfo.jsx:3]: The React component import for `AdditionalInfo` can be removed if you just use the `va-additional-info` web component below.

[the remark this answers is not in the document]
Comment (adamwhitlock1) [reply script/component-migration/samples/AdditionalInfo.jsx:3]: @jamigibbs I thought this was a script to provide an example of how the migration script would replace a legacy react component with the web-component? I don't even know if this is even a currently used piece of code in reality.

[the remark this answers is not in the document]
Comment (jamigibbs) [reply script/component-migration/samples/AdditionalInfo.jsx:3]: oh! I see. Carry on then! 🙃

**Review by jamigibbs — APPROVED** [review]

File: script/component-migration/samples/AdditionalInfo.jsx (hunk 1 above)
Comment (adamwhitlock1) [reply script/component-migration/samples/AdditionalInfo.jsx:76]: See previous comment, but I think this is supposed to be a sample of a legacy implementation with the React component, that you could then run the migration script on to migrate to the web component version? 🤷🏼

**Pull request merged**